internal/service/dashboards.go

```diff
@@ -28,6 +32,110 @@ type DashboardsApi interface {
 	DownloadDashboards(filter filters.Filter) []string
 	UploadDashboards(filter filters.Filter)
 	DeleteAllDashboards(filter filters.Filter) []string
+	LintDashboards(req LintRequest) []string
+}
+
+type LintRequest struct {
+	StrictFlag    bool
+	VerboseFlag   bool
+	AutoFix       bool
+	DashboardSlug string
+	FolderName    string
+}
+
+func (s *DashNGoImpl) LintDashboards(req LintRequest) []string {
+	var (
+		rawBoard []byte
+	)
+	dashboardPath := config.Config().GetDefaultGrafanaConfig().GetPath(config.DashboardResource)
+	filesInDir, err := s.storage.FindAllFiles(dashboardPath, true)
+	if err != nil {
+		log.Fatalf("unable to find any files to export from storage engine, err: %v", err)
+	}
+	filterReq := NewDashboardFilter(req.FolderName, req.DashboardSlug, "")
+	validFolders := filterReq.GetEntity(filters.FolderFilter)
+	for _, file := range filesInDir {
+		baseFile := filepath.Base(file)
+		baseFile = strings.ReplaceAll(baseFile, ".json", "")
+
+		if !strings.HasSuffix(file, ".json") {
+			slog.Warn("Only json files are supported, skipping", "filename", file)
+			continue
+		}
+		if req.DashboardSlug != "" && baseFile != req.DashboardSlug {
+			slog.Debug("Skipping dashboard, does not match filter", slog.String("dashboard", req.DashboardSlug))
+			continue
+		}
+
+		if rawBoard, err = s.storage.ReadFile(file); err != nil {
+			slog.Warn("Unable to read file", "filename", file, "err", err)
+			continue
+		}
+		if req.FolderName != "" {
+			if !slices.Contains(validFolders, req.FolderName) && !config.Config().GetDefaultGrafanaConfig().GetFilterOverrides().IgnoreDashboardFilters {
+				slog.Debug("Skipping file since it doesn't match any valid folders", "filename", file)
+				continue
+			}
+		}
+
+		dashboard, err := lint.NewDashboard(rawBoard)
+		if err != nil {
+			slog.Error("failed to parse dashboard", slog.Any("err", err))
+			continue
+		}
+		lintConfigFlag := strings.ReplaceAll(file, ".json", ".lint")
+		cfgLint := lint.NewConfigurationFile()
+		if err := cfgLint.Load(lintConfigFlag); err != nil {
+			slog.Error("Unable to load lintConfigFlag")
+			continue
+		}
+		cfgLint.Verbose = req.VerboseFlag
+		cfgLint.Autofix = req.AutoFix
+
+		rules := lint.NewRuleSet()
+		results, err := rules.Lint([]lint.Dashboard{dashboard})
+		if err != nil {
+			slog.Error("failed to lint dashboard", slog.Any("err", err))
+			continue
+
+		}
+		if cfgLint.Autofix {
+			changes := results.AutoFix(&dashboard)
+			if changes > 0 {
+				slog.Info("AutoFix possible")
+				writeErr := s.writeLintedDashboard(dashboard, file, rawBoard)
+				if writeErr != nil {
+					slog.Error("unable to autofix linting issues for dashboard", slog.String("dashboard", file))
+				}
+			} else {
+				slog.Error("AutoFix is not possible for dashboard.", slog.String("dashboard", file))
+				time.Sleep(time.Second * 5)
```

Comment:
Why the sleep here? This doesn't feel consistent with other error reporting.

Reply:
The output is pretty extensive.  I'm just pausing a bit to show the error before continuing as it's not critical.  It'll get buried otherwise.